### DISCUSSION TITLE
osu!lazer taiko SV changing drastically based off of screen scaling changes

### DISCUSSION THREAD

**Discussion by zundazun** [ppy/osu#31253] [Q&A]
### Type

Game behaviour

### Bug description

Using the screen scaling settings changes how the slider velocity is displayed in a map in taiko. Videos 1-2 display vertical scaling on a low SV map (https://osu.ppy.sh/beatmapsets/1963076#taiko/4069626). Video 3 displays horizontal scaling and how low horizontal scaling shows less notes (https://osu.ppy.sh/beatmapsets/1975566#taiko/4100176).

Playing with low vertical scaling speeds up notes while also displaying more notes. Additionally playing with low horizontal scaling does not seem to change the speed, but instead effectively crops out the amount of notes you would see if you were playing in 16:9 instead.

Because of these behaviors, some maps that use slow SV are significantly easier with low vertical scaling, and players that use resolutions such as 4:3 will inherently always have a disadvantage on every map (this is not the case on stable).

I checked to see if this has been reported anywhere but I wasn't able to find anything on this, which is surprising, unless I'm just blind.

### Screenshots or videos

https://github.com/user-attachments/assets/b80303cb-3158-4103-934c-be66bbe62fb3

https://github.com/user-attachments/assets/ace3055b-b3b8-400a-9c9e-dec9d89e99b2

https://github.com/user-attachments/assets/9b22ba2e-1733-4a49-85c8-8af4dbd0b712

### Version

2024.1219.2-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/18233783/compressed-logs.zip)

```[tasklist]
### Tasks
```

**Comment by bdach** [comment]
Yes this happens, yes we know of it, it has even been called an "exploit" in the past. This is a result of the taiko stupidity around screen aspect ratio and what should happen on extremely thin screens (show more notes or space them out more). We've flip-flopped on this issue several times and are pretty done talking about it because it's stupid. The original lazer solution that worked was fixing playfield aspect ratio to 16:9 always, but people hated it and wanted 4:3 to cut off their screens or whatever which is where all of that aspect ratio dumbness started from.

If the taiko community at large can agree on what the proper solution to this sort of thing is then we can implement it. But I'm generally not willing to touch this any further just off of "hey this is bad" without a constructive counterargument what should actually happen if you do things like this.

**Reply by bdach** [reply]
> Because of these behaviors, some maps that use slow SV are significantly easier with low vertical scaling, and players that use resolutions such as 4:3 will inherently always have a disadvantage on every map (this is not the case on stable).

Oh and I don't know where this is coming from either. 4:3 and 16:9 SV should match. None of the screenshots you shared corroborate it not matching.
